backend/entityservice/tasks/comparing.py

```diff
@@ -208,7 +212,65 @@ def compute_filter_similarity(chunk_info_dp1, chunk_info_dp2, project_id, run_id
         t6 - t5,
         t6 - t0)
     )
-    return num_results, result_filename
+    return num_results, file_size, result_filename
+
+
+def _put_placeholder_empty_file(mc, log):
+    sims = array.array('d')
+    dset_is0 = array.array('I')
+    dset_is1 = array.array('I')
+    rec_is0 = array.array('I')
+    rec_is1 = array.array('I')
+    candidate_pairs = sims, (dset_is0, dset_is1), (rec_is0, rec_is1)
+    empty_file_iter, empty_file_size \
+        = anonlink.serialization.dump_candidate_pairs_iter(candidate_pairs)
+    empty_file_name = Config.SIMILARITY_SCORES_FILENAME_FMT.format(
+            generate_code(12))
+    empty_file_stream = iterable_to_stream(empty_file_iter)
+    try:
+        mc.put_object(Config.MINIO_BUCKET, empty_file_name,
+                      empty_file_stream, empty_file_size)
+    except minio.ResponseError:
+        log.warning("Failed to store empty result in minio.")
+        raise
+    return 0, empty_file_size, empty_file_name
+
+
+def _merge_files(mc, log, file0, file1):
+    num0, filesize0, filename0 = file0
+    num1, filesize1, filename1 = file1
+    total_num = num0 + num1
+    file0_stream = mc.get_object(Config.MINIO_BUCKET, filename0)
+    file1_stream = mc.get_object(Config.MINIO_BUCKET, filename1)
+    merged_file_iter, merged_file_size \
+        = anonlink.serialization.merge_streams_iter(
+            (file0_stream, file1_stream), sizes=(filesize0, filesize1))
+    merged_file_name = Config.SIMILARITY_SCORES_FILENAME_FMT.format(
+            generate_code(12))
+    merged_file_stream = iterable_to_stream(merged_file_iter)
+    try:
+        mc.put_object(Config.MINIO_BUCKET, merged_file_name,
```

Comment:
Are they partial merge files deleted after merge is complete?

Reply:
They’re deleted on [line 257](#diff-30fb0765378c9296147bce2d345daed9R257) :)